spyder/utils/introspection/manager.py

```diff
@@ -154,7 +183,18 @@ def _handle_timeout(self):
         self.waiting = False
         if self.pending:
             self._finalize(self.pending)
+        elif self.pending_request:
+            info = self.pending_request
+            self.pending_request = None
+            self.send_request(info)
         else:
+            self.timeout_counter += 1
+            if (self.timeout_counter > MAX_TIMEOUTS and
+                    not self.plugins_initialized['jedi']):

```

Comment:
Could you explain this here? In particular:

1. What's the meaning of MAX_TIMEOUTS? Does it mean that the user has to send MAX_TIMEOUTS petitions to the server before it's restarted?
2. Why are you looking for Jedi here? Is Jedi the only one that's giving troubles?